app/Notifications/NewPost.php

```diff
@@ -45,21 +45,28 @@ public function via(object $notifiable): array
 
     /**
      * Determine if the notification should be sent.
-     *
-     * @return bool
      */
     public function shouldSend(User $notifiable): bool
     {
         $targetGroup = '';
 
-        if ($this->type == 'subscription') {
-            $targetGroup = 'json_subscription_groups';
-        } elseif ($this->type == 'staff') {
-            $targetGroup = '';
-        } elseif ($this->type == 'topic') {
-            $targetGroup = 'json_forum_groups';
-        } else {
-            $targetGroup = 'json_forum_groups';
+        switch ($this->type) {
+            case 'subscription':
+                $targetGroup = 'json_subscription_groups';
+
+                break;
+            case 'staff':
+                $targetGroup = '';
+
+                break;
+            case 'topic':
+                $targetGroup = 'json_forum_groups';
+
+                break;
+            default:
+                $targetGroup = 'json_forum_groups';
+
+                break;
```

Comment:
Can this code maybe be inlined below? Something like the following:

```php
return match ($this->type) {
    'subscription' => !\in_array($this->user->group->id, $notifiable->notification->json_subscription_groups, true);
    'staff'        => true;
    default        => !\in_array($this->user->group->id, $notifiable->notification->json_forum_groups, true);
}
```

This should make the code a lot cleaner (once the factory's array property is fixed)